velox/connectors/hive/TableHandle.h

```diff
@@ -30,14 +30,30 @@ class HiveColumnHandle : public ColumnHandle {
  public:
   enum class ColumnType { kPartitionKey, kRegular, kSynthesized };
 
+#ifdef VELOX_ENABLE_BACKWARD_COMPATIBILITY
   HiveColumnHandle(
       const std::string& name,
       ColumnType columnType,
       TypePtr dataType,
       std::vector<common::Subfield> requiredSubfields = {})
+      : HiveColumnHandle(
+            name,
+            columnType,
+            std::move(dataType),
+            nullptr,
+            std::move(requiredSubfields)) {}
+#endif
+
+  HiveColumnHandle(
+      const std::string& name,
+      ColumnType columnType,
+      TypePtr dataType,
+      TypePtr hiveType,
```

Comment:
Would you add comments to explain the difference between dataType and hiveType and clarify when these would be different?

Reply:
Added. Thanks!